pkgs/by-name/ba/bambu-unwrapped/package.nix

```diff
@@ -0,0 +1,361 @@
+) clangFrontends;
+assert builtins.all (
+  gcc:
+  lib.assertMsg (
+    gcc.stdenv.hostPlatform.system == "i686-linux"
+  ) "Bambu frontends have to be i686 but ${gcc.name} is ${gcc.stdenv.hostPlatform.system}"
+) gccFrontends;
+
+# Assert that the clang and GCC frontends versions are compatible with bambu
+assert builtins.all (
+  llvmPackages:
+  lib.assertMsg (builtins.elem (lib.versions.major llvmPackages.clang.version) supportedLlvmMajorVersions) "Clang version ${llvmPackages.clang.version} is not supported as a frontend by bambu. The following major clang versions are supported: ${builtins.concatStringsSep " " supportedLlvmMajorVersions}"
+) clangFrontends;
+assert builtins.all (
+  gcc:
+  lib.assertMsg (builtins.elem (lib.versions.major gcc.version) supportedGccMajorVersions) "GCC version ${gcc.version} is not supported as a frontend by bambu. The following major GCC versions are supported: ${builtins.concatStringsSep " " supportedGccMajorVersions}"
+) gccFrontends;
+
+# Assert that there is only one compiler frontend for each major version of the compiler
+let
+  passedClangVersions = builtins.map (llvmPackages: llvmPackages.clang.version) clangFrontends;
+  duplicateClangVersion = lib.lists.findFirst (
+    clangVersion:
+    (builtins.length (
+      builtins.filter (
+        otherClangVersion: (lib.versions.major clangVersion) == (lib.versions.major otherClangVersion)
+      ) passedClangVersions
+    )) > 1
+  ) null passedClangVersions;
+  passedGccVersions = builtins.map (gcc: gcc.version) gccFrontends;
+  duplicateGccVersion = lib.lists.findFirst (
+    gccVersion:
+    (builtins.length (
+      builtins.filter (
+        otherGccVersion: (lib.versions.major gccVersion) == (lib.versions.major otherGccVersion)
+      ) passedGccVersions
+    )) > 1
+  ) null passedGccVersions;
+in
+assert lib.assertMsg (duplicateClangVersion == null)
+  "Multiple clang frontends with the same major version (${lib.versions.major duplicateClangVersion}) are not supported by bambu. You passed packages with the following versions: ${
+    builtins.concatStringsSep " " (
+      builtins.filter (
+        clangVersion: (lib.versions.major clangVersion) == (lib.versions.major duplicateClangVersion)
+      ) passedClangVersions
+    )
+  }";
+
+assert lib.assertMsg (duplicateGccVersion == null)
+  "Multiple gcc frontends with the same major version (${lib.versions.major duplicateGccVersion}) are not supported by bambu. You passed packages with the following versions: ${
+    builtins.concatStringsSep " " (
+      builtins.filter (
+        gccVersion: (lib.versions.major gccVersion) == (lib.versions.major duplicateGccVersion)
+      ) passedGccVersions
+    )
+  }";
+
+gcc12Stdenv.mkDerivation rec {
+  pname = "bambu-unwrapped";
+  version = "2024.03";
+
+  src = fetchFromGitHub {
+    owner = "ferrandi";
+    repo = "PandA-bambu";
+    rev = "v${version}";
+    hash = "sha256-TPcVrQqk2SHHTYrBwpl1f8YM5qONEpRwcAetmFAfFWM=";
+    leaveDotGit = true;
```

Comment:
```suggestion
```
Instead, patch the relevant part

https://github.com/ferrandi/PandA-bambu/blob/aca8437607356c02a9501c89da414dfe8f4e5fc8/src/Makefile.am#L14-L19